WeartGloves/src/WeArtGlove.cpp

```diff
@@ -0,0 +1,1598 @@
+        indexGravityParameterHandler->setParameter("kp", convergenceRate);
+        const Eigen::Vector2d indexWeightRegularizationGr = 10 * Eigen::Vector2d::Ones(2);
+        indexGravityParameterHandler->setParameter("weight", indexWeightRegularizationGr);
+        indexGravityParameterHandler->setParameter("target_frame_name", fingerTipFrameNameList[1]);
+        indexGravityParameterHandler->setParameter("priority", 1);
+        parameterHandler->setGroup("INDEX_GRAVITY_TASK", indexGravityParameterHandler);
+
+        // middle
+        auto middleGravityParameterHandler = std::make_shared<StdImplementation>();
+        middleGravityParameterHandler->setParameter("type", "GravityTask");
+        middleGravityParameterHandler->setParameter("kp", convergenceRate);
+        const Eigen::Vector2d middleWeightRegularizationGr = 10 * Eigen::Vector2d::Ones(1);
+        middleGravityParameterHandler->setParameter("weight", middleWeightRegularizationGr);
+        middleGravityParameterHandler->setParameter("target_frame_name", fingerTipFrameNameList[2]);
+        middleGravityParameterHandler->setParameter("priority", 1);
+        parameterHandler->setGroup("MIDDLE_GRAVITY_TASK", middleGravityParameterHandler);
+
+        // Construct the IK
+        ikProblem = std::make_shared<IntegrationBasedIKProblem>(QPInverseKinematics::build(parameterHandler, kinDyn));
+
+        auto baseSE3Task = std::dynamic_pointer_cast<SE3Task>(ikProblem->ik->getTask("SE3_TASK").lock());
+        baseSE3Task->setTaskControllerMode(SE3Task::Mode::Disable);
+        baseSE3Task->setSetPoint(manif::SE3d::Identity(),
+                                 manif::SE3d::Tangent::Zero());
+
+        auto regularizationTask = std::dynamic_pointer_cast<JointTrackingTask>(ikProblem->ik->getTask("REGULARIZATION_TASK").lock());
+        regularizationTask->setSetPoint(regularizationRate * Eigen::VectorXd::Ones(kinDyn->model().getNrOfDOFs()));
+
+        // Distance Task
+        thumbDistanceTaskptr = std::dynamic_pointer_cast<DistanceTask>(ikProblem->ik->getTask("THUMB_DISTANCE_TASK").lock());
+
+        indexDistanceTaskptr = std::dynamic_pointer_cast<DistanceTask>(ikProblem->ik->getTask("INDEX_DISTANCE_TASK").lock());
+
+        middleDistanceTaskptr = std::dynamic_pointer_cast<DistanceTask>(ikProblem->ik->getTask("MIDDLE_DISTANCE_TASK").lock());
+
+        // Gravity Task
+        palmGravityTaskptr = std::dynamic_pointer_cast<GravityTask>(ikProblem->ik->getTask("PALM_GRAVITY_TASK").lock());
+
+        thumbGravityTaskptr = std::dynamic_pointer_cast<GravityTask>(ikProblem->ik->getTask("THUMB_GRAVITY_TASK").lock());
+
+        indexGravityTaskptr = std::dynamic_pointer_cast<GravityTask>(ikProblem->ik->getTask("INDEX_GRAVITY_TASK").lock());
+
+        middleGravityTaskptr = std::dynamic_pointer_cast<GravityTask>(ikProblem->ik->getTask("MIDDLE_GRAVITY_TASK").lock());
+
+        jointVelocity = Eigen::VectorXd::Zero(model.getNrOfDOFs());
+
+    }
```

Comment:
I would add an else case printing an error mentioning that the tracking algorithm is not supported